packages/react/src/components/Forms/DatePicker/DateRangePicker.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useMemo, useRef, useState } from "react";
+import { DateValue } from "@internationalized/date";
+import {
+  DateRangePickerStateOptions,
+  useDateRangePickerState,
+} from "@react-stately/datepicker";
+import { useDateRangePicker } from "@react-aria/datepicker";
+import { useDateFormatter } from "@react-aria/i18n";
+import { CalendarRange } from ":/components/Forms/DatePicker/Calendar";
+import DatePickerAux, {
+  DatePickerAuxSubProps,
+} from ":/components/Forms/DatePicker/DatePickerAux";
+import DateFieldBox from ":/components/Forms/DatePicker/DateField";
+import { StringsOrDateRange } from ":/components/Forms/DatePicker/types";
+import {
+  getDefaultPickerOptions,
+  parseRangeCalendarDate,
+} from ":/components/Forms/DatePicker/utils";
+
+type DateRangePickerProps = DatePickerAuxSubProps & {
+  startLabel: string;
+  endLabel: string;
+  value?: null | StringsOrDateRange;
+  defaultValue?: StringsOrDateRange;
+  onChange?: (value: [string, string] | null) => void | undefined;
+};
+
+const DateRangePicker = ({
+  startLabel,
+  endLabel,
+  ...props
+}: DateRangePickerProps) => {
+  if (props.defaultValue && props.value) {
+    throw new Error(
+      "You cannot use both defaultValue and value props on DateRangePicker component"
+    );
+  }
+  const ref = useRef<HTMLDivElement>(null);
+  const [isFocused, setIsFocused] = useState(false);
+  const formatter = useDateFormatter();
+
+  const options: DateRangePickerStateOptions<DateValue> = {
+    ...getDefaultPickerOptions(props),
+    value: props.value === null ? null : parseRangeCalendarDate(props.value),
+    defaultValue: parseRangeCalendarDate(props.defaultValue),
+    onChange: (value: any) => {
+      props.onChange?.(
+        value?.start && value.end
+          ? [value.start.toString(), value.end.toString()]
+          : null
+      );
+    },
+  };
+  const pickerState = useDateRangePickerState(options);
+  const { startFieldProps, endFieldProps, calendarProps, ...pickerProps } =
+    useDateRangePicker(options, pickerState, ref);
+
+  const labelAsPlaceholder = useMemo(
+    () =>
+      !pickerState.isOpen &&
+      (!pickerState.value.start || !pickerState.value.end),
```

Comment:
invert condition, if at least one is filled, `labelAsPlaceholder` should be false.